en_US/faq/use-guide.md

```diff
@@ -417,3 +417,141 @@ Client disconnect link error code list:
+
+### Keyword: “reason: {shutdown, discarded}”
+
+Contrary to session takeover, if the client indicates that it wants to start a new session when connecting, then EMQX needs to discard the old session state, and EMQX will close the old process with the reason `{shutdown, discarded}`. If the original client is still connected, EMQX will send a DISCONNECT packet with Reason Code 0x8E (Session taken over), and then close the old connection.
+
+### Keyword: “reason: {shutdown, kicked}”
+
+Indicates that the client connection was manually kicked out (click the Kick Out button on the Dashboard or call the `DELETE clients/{clientid}` API). If the original client is still connected, EMQX will send a DISCONNECT packet with Reason Code 0x98 (Administrative action), and then close the old connection.
+
+### Keyword: “reason: {shutdown, tcp_closed}”
+
+This log indicates that the client directly closed the network connection without sending a DISCONNECT packet. The corresponding reason would be `{shutdown, normal}`, which means that the client first sent a DISCONNECT packet with a Reason Code of 0 and then closed the network connection.
+
+Therefore, when you encounter that `tcp_closed` appears in the EMQX logs, and this disconnection behavior is not what you expected, it is recommended to investigate the client's implementation for any potential issues.
+
+### Keyword: “maximum heap size reached”
+
+The appearance of this log indicates that the heap occupied by the current client process has exceeded the preset maximum value, and the client process will be forcibly killed to ensure the availability of EMQX, and avoid the unlimited growth of the memory usage of the client process and eventually lead to EMQX OOM. 
+
+Usually, message accumulation is the main reason for the increase in client process heap occupation, and message accumulation is generally because the client's consumption capacity does not match the peer's message production capacity. The recommended solution is to optimize the client processing code or use shared subscriptions to disperse the load.
+
+The related configuration item is `force_shutdown_policy`, and its format is `<Maximum Message Queue Length>|<Maximum Heap Size>`, for example, `10000|64MB`.  `<Maximum Heap Size>` refers to the maximum heap memory that each client process can occupy.
+
+### Keyword: “Parse failed for function_clause”
+
+This log indicates that the parsing of the packet failed. Maybe because it is not an MQTT packet, we have encountered many situations where HTTP requests are sent to the MQTT port, or because the packet contains non-UTF-8 characters. We can search the `Frame data` keyword in this "Parse failed..." log to view the complete message and help us analyze the possible reasons for the parsing failure.
```

Comment:
```suggestion
This log indicates that the parsing of the packet failed. Maybe because it is not an MQTT packet, we have encountered many situations where HTTP requests are sent to the MQTT port, or because the packet contains non-UTF-8 characters. We can search the `Frame data` keyword in this "Parse failed..." log to view the complete message and help us analyze the possible reasons for the parsing failure.
```
```suggestion
This log indicates a failure in packet parsing. It may occur when a non-MQTT packet is sent to the MQTT port, or when the packet contains non-UTF-8 characters. Searching for the `Frame data` keyword in the `Parse failed...` log can provide the complete message and help analyze the possible reasons for the parsing failure.
```